### PR TITLE
Update Omaha dependencies in brave://credits

### PR DESCRIPTION
**Pull request by fmarier**
Resolves https://github.com/brave/brave-browser/issues/35743

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

(Windows-only)

1. Open `brave://credits`.
2. Ensure that the following entries exist and have working _show license_ and _homepage_ links:
   - LZMA SDK
   - libzip
   - Omaha
   - zlib

### CODE REVIEW

**Review by emerick — APPROVED** [review]
LGTM

File: script/brave_license_helper.py [hunk 1]
```diff
@@ -38,6 +38,34 @@ def AddBraveCredits(root, prune_paths, special_cases, prune_dirs,
         os.path.join('brave', 'third_party', 'libaddressinput'),
         os.path.join('brave', 'patches', 'third_party'),
 
+        # Duplicate notices already included in brave-core notices
```
Comment (mherrmann) [inline script/brave_license_helper.py:41]: This comment confused me at first. Can we rephrase it to:
```suggestion
        # Dependencies that are already in brave-core, and whose notices
        # therefore do not need to be repeated.
```

File: script/brave_license_helper.py [hunk 2]
```diff
@@ -38,6 +38,34 @@ def AddBraveCredits(root, prune_paths, special_cases, prune_dirs,
...
+        os.path.join('brave', 'vendor', 'omaha', 'omaha', 'third_party',
+                     'chrome'),
+        os.path.join('brave', 'vendor', 'omaha', 'third_party', 'libzip'),
+        os.path.join('brave', 'vendor', 'omaha', 'third_party', 'lzma'),
+        os.path.join('brave', 'vendor', 'omaha', 'third_party', 'zlib'),
+
+        # Dependencies already included in the main breakpad notice.
```
Comment (mherrmann) [inline script/brave_license_helper.py:48]:
```suggestion
        # Dependencies already mentioned in the main breakpad notice.
```

File: script/brave_license_helper.py [hunk 3]
```diff
@@ -38,6 +38,34 @@ def AddBraveCredits(root, prune_paths, special_cases, prune_dirs,
...
+                     'src', 'third_party', 'curl'),
+        os.path.join('brave', 'vendor', 'omaha', 'third_party', 'breakpad',
+                     'src', 'third_party', 'libdisasm'),
+        os.path.join('brave', 'vendor', 'omaha', 'third_party', 'breakpad',
+                     'src', 'third_party', 'linux'),
+        os.path.join('brave', 'vendor', 'omaha', 'third_party', 'breakpad',
+                     'src', 'third_party', 'mac_headers'),
+
+        # Dependencies which are no longer present in the codebase. This should
+        # be cleaned up in the upstream dependency including this.
```
Comment (mherrmann) [inline script/brave_license_helper.py:59]:
```suggestion
        # Essentially empty directories.
```

File: script/brave_license_helper.py [hunk 4]
```diff
@@ -38,6 +38,34 @@ def AddBraveCredits(root, prune_paths, special_cases, prune_dirs,
...
+        os.path.join('brave', 'vendor', 'omaha', 'omaha', 'third_party',
+                     'hashlib'),
+
+        # No licensing information in recursive dependency. This should be fixed
+        # by the upstream dependency including this.
+        os.path.join('brave', 'vendor', 'omaha', 'omaha', 'third_party', 'bar'),
```
Comment (mherrmann) [inline script/brave_license_helper.py:65]: The `bar/` directory does not exist for me. Was this line added by accident?

File: script/brave_license_helper.py [hunk 5]
```diff
@@ -38,6 +38,34 @@ def AddBraveCredits(root, prune_paths, special_cases, prune_dirs,
...
+        os.path.join('brave', 'vendor', 'omaha', 'third_party', 'breakpad',
+                     'src', 'third_party', 'mac_headers'),
+
+        # Dependencies which are no longer present in the codebase. This should
+        # be cleaned up in the upstream dependency including this.
+        os.path.join('brave', 'vendor', 'omaha', 'omaha', 'third_party',
+                     'hashlib'),
+
+        # No licensing information in recursive dependency. This should be fixed
+        # by the upstream dependency including this.
```
Comment (mherrmann) [inline script/brave_license_helper.py:64]:
```suggestion
        # No licensing information in recursive dependency.
```

**Review by mherrmann — CHANGES_REQUESTED** [review]

File: script/brave_license_helper.py (hunk 4 above)
Comment (fmarier) [reply script/brave_license_helper.py:65]: You're right, it looks like I hadn't moved my local repo to the new `main` branch (I was still on `master`). That's where `bar` was. I'll take it out.

File: script/brave_license_helper.py (hunk 5 above)
Comment (fmarier) [reply script/brave_license_helper.py:64]: I do want to keep this one because in general "No licensing info" is no excuse for not including it.

Comment (mherrmann) [reply script/brave_license_helper.py]: Ok, can we then rephrase it as `No licensing information in recursive dependency. This should be fixed by the upstream dependency`?

Comment (fmarier) [reply script/brave_license_helper.py]: That extra comment is there to try and avoid developers seeing this as an easy way to ignore build failures due to the licensing checks. :crossed_fingers:

Comment (mherrmann) [reply script/brave_license_helper.py]: I'm just confused what "this" refers to here, in the second half of the sentence. Is it the same thing that's meant by the first "this" in the sentence?

Comment (fmarier) [reply script/brave_license_helper.py]: Good point. I've changed the wording to make it clearer. Removing the double use of "this".
